### ISSUE TITLE
Missing Wifi information

### ISSUE THREAD

**Issue by sdrapha** [JanDeDobbeleer/oh-my-posh#4238]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

Today I did an update, and then found out my Wifi segment completely vanished. It took me a good amount of time troubleshooting it, until I've found #2857.
There was no migration path, or deprecation warning, not cool.

Also, after I migrated my configuration to the new `connection` segment, there are still missing information:
`.Signal` `.ReceiveRate` are not working

### Theme

my own `my-posh-theme.omp.yaml` which was probably based on some default powerline styled theme, can't remember now. 

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

powershell

### Log output

```Shell
-
```

**Comment by JanDeDobbeleer** [comment]
> There was no migration path, or deprecation warning, not cool.

All major versions have their breaking changes added to the release notes:

- https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v12.0.0
- https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v13.0.0
- https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v14.0.0
- https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v15.0.0
- https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v16.0.0
- https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v17.0.0
- https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v18.0.0

You have all of these increments to look at and adjust if needed. Have a look at the [new documentation](https://ohmyposh.dev/docs/segments/connection) to see what's still supported.

If we can make a migration, we will, but in this case a manual intervention is required.

**Comment by sdrapha** [comment]
To be fair, I updated the package through a package manager, in this instance `scoop`, sot I didn't see the github notes. I guess I was expecting an in-prompt warning, like swapping the old wifi segment with a deprecation message instead of the actual info. But okay, I get it.

What about the other part of the issue, I found out that `.ReceiveRate` actually does work, but it's [undocumented](https://ohmyposh.dev/docs/segments/connection#template-info).

And then, about `.Signal` is there no plans to bring that back?
Would that be a feature request based on the current latest code?
![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/25059996/905b947c-ad17-4063-81a7-35f92b715e67)
I see there is a glimpse of code in that direction, but not much more implementation around that.

Anyway, I forgot to say on the first post, thanks for the awesome work with the whole project, appreciate it a lot.
